include/debug/coredump.h

```diff
@@ -136,6 +170,30 @@ void z_coredump_buffer_output(uint8_t *buf, size_t buflen)
  * @param buflen Buffer length
  */
 
+/**
+ * @fn int z_coredump_query(enum z_coredump_query_id query_id, void *arg);
```

Comment:
are those public APIs? if yes, we should not be using z_, z_ is only for private APIs used within the subsystem itself, not applicaton or user facing.

Reply:
My thinking is that these are not to be used as-is, and will always be used under another layer of subsystem (e.g. telemetry). That's why these are not public facing APIs.

Reply:
ok, we should remove z_ prefix. z_ is basically a replacement for plain _ and signifies a private function within a subsystem. This is a public header and the APIs have documentation and will be used to add new backends and such, so it is not private and defines an interface. Even if not application facing.

Reply:
Removed the z_ prefix in public APIs.